src/components/ProductTours/data/utils.js

```diff
@@ -0,0 +1,41 @@
+// Enable all cookies when onDismiss is called to ensure that the tour is not shown again
+import Cookies from 'universal-cookie';
+import {
+  COOKIE_NAMES,
+  PORTAL_APPEARANCE_TOUR_COOKIE_NAME,
+  BROWSE_AND_REQUEST_TOUR_COOKIE_NAME,
+  LEARNER_CREDIT_COOKIE_NAME,
+  HIGHLIGHTS_COOKIE_NAME,
+} from '../constants';
+
+const cookies = new Cookies();
+
+export function filterCheckpoints(checkpoints, enabledFeatures) {
```

Comment:
I believe this function could be simplified considerably? `enabledFeatures` is already a map of which checkpoints should be enabled, based on the presence of the related cookies for each checkpoint, so the filtering based on cookie names that already exist is duplicative effort. I think you could simply rely on filtering `checkpoints` based on the `enabledFeatures` directly, e.g.:

```jsx
export function filterCheckpoints(checkpoints, enabledFeatures) {
  const filteredCheckpoints = [];
  Object.keys(checkpoints).forEach((checkpoint) => {
    if (enabledFeatures[checkpoint]) {
      filteredCheckpoints.push(checkpoints[checkpoint]);
    }
  });

  return filteredCheckpoints;
}
```